### PR TITLE
Filter Redirect URL for Cloned Posts

### PR DESCRIPTION
**Pull request by GFargo**
Adds two new filters that allow developers to modify the resulting redirect URL.  This PR bumps the plugin version to `0.2` and adds a note to `readme.txt` about the new filter.

### CODE REVIEW

File: class-cr-clone.php [hunk 1]
```diff
@@ -64,7 +64,17 @@ public function __action_admin_post() {
 			wp_die( esc_html__( 'There was an error copying this post', 'clone-replace' ) );
 		}
 
-		wp_redirect( admin_url( "post.php?post={$post_id}&action=edit" ) );
+		// Modify individual query args for redirect URL.
+		$admin_redirect_url = add_query_arg(
+			apply_filters( 'CR_Clone_redirect_query_args', [
```
Comment (mboynes) [inline class-cr-clone.php]: Filters need phpdoc per [inline documentation standards](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/) (also applies to line 77).

File: class-cr-clone.php [hunk 2]
```diff
@@ -64,7 +64,17 @@ public function __action_admin_post() {
...
+				'post'   => $post_id,
+				'action' => 'edit',
+			] ),
+			admin_url( "post.php" )
+		);
+
+		// Change redirect url completely using the filter below.
+		wp_redirect( esc_url( apply_filters( 'CR_Clone_redirect_url', $admin_redirect_url ) ) );
```
Comment (mboynes) [inline class-cr-clone.php]: `esc_url()` should only be used when _outputting_ a URL in HTML, e.g. an `href` or `src` attribute in an HTML tag, as it encodes entities. `esc_url_raw()` should be used to escape a URL in every other circumstance (redirects, outputting in JavaScript, REST API responses, etc.)

File: class-cr-clone.php [hunk 3]
```diff
@@ -64,7 +64,17 @@ public function __action_admin_post() {
...
 		}
 
-		wp_redirect( admin_url( "post.php?post={$post_id}&action=edit" ) );
+		// Modify individual query args for redirect URL.
+		$admin_redirect_url = add_query_arg(
+			apply_filters( 'CR_Clone_redirect_query_args', [
+				'post'   => $post_id,
+				'action' => 'edit',
+			] ),
+			admin_url( "post.php" )
```
Comment (mboynes) [inline class-cr-clone.php]: single quotes

**Comment by GFargo** [comment]
Sure thing.  By modifying the redirect I will be able to add a URL parameter that can be detected and used to add admin notices on page load.  When writing this I was thinking that appending an item to the array would be cleaner than modifying the URL string 🤷‍♂️ 

So originally the first filter was created for simplicity, but this could be accomplished using only the second filter.  I'll remove it and address the CR comments.

File: class-cr-clone.php [hunk 4]
```diff
@@ -64,7 +64,18 @@ public function __action_admin_post() {
...
+		/**
+		 * Filter Cloned Post Redirect URL.
+		 *
+		 * Modify the resulting redirect location after cloning a post.
+		 *
+		 * @since 0.2
+		 *
+		 * @param string $redirect_url URL string passed to wp_redirect
+		 */
+		$redirect_url = apply_filters( 'CR_Clone_redirect_url', admin_url( "post.php?post={$post_id}&action=edit" ) );
```
Comment (mboynes) [inline class-cr-clone.php]: Two more little things:
* [ ] Pass the post id, so other code hooking into this doesn't have to parse it from the url (don't forget to document the `@param` too)
* [ ] This line is too long, so break it up, e.g.
   ```php
   apply_filters(
      'CR_Clone_redirect_url',
      admin_url( ... ),
      $post_id
   );
   ```

**Pull request merged**